runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```
@@ -0,0 +1,277 @@
+        }
+    }
+
+    /**
+     * Remove destination from LogReplicationSourceMetadataTable
+     *
+     * @param logicalGroup String representative of the group of tables which are replicated
+     * @param destination Cluster from which the logicalGroup should be removed from further replication
+     * @return true if destination was removed
+     */
+    public boolean removeDestination(String logicalGroup, String destination) {
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+
+                    ClientInfoKey clientInfoKey = ClientInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        SinksInfoVal clientDestinationIds = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinationIds.getDestinationIdsList());
+
+                        if (clientDestinationIdsList.remove(destination)) {
+                            clientDestinationIds = clientDestinationIds.toBuilder()
+                                    .clearDestinationIds()
+                                    .addAllDestinationIds(clientDestinationIdsList)
+                                    .build();
+                            txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinationIds, null);
+                            txn.commit();
+                            return true;
+                        }
+                        return false;
+                    }
+                    return false;
```

I think we should throw an exception here saying record not found.